File: Lenguajes_Programacion/10_Kotlin/1_Sintaxis_Kotlin/1_introduccion.kt
// Lenguaje Kotlin
// Fue creado en la empresa Jetbrains
// - Es fuertemente tipado, pero puede inferir el dato

package 1_Sintaxis_Kotlin

fun main() {

    /* 
    * Comentario.
    */

    // Hola mundo
    println("Hola mundo")

    // - Variables

    // 1. Variable de tipo cadena (String)

    var myString = "Cadena de texto"
    myString = "Cambiando el valor de un texto"
    // myString = 6 // Error No se puede cambiar el valor que sea de otro tipo de dato que no sea el asignado.
    println(myString)

    var myString2: String = "Esta es otra cadena de texto"
    println(myString2)

    // 2. Variable de tipo entero (Int, etc)

    var myInt = 7
    myInt = myInt + 4
    println(myInt) // 11
    println(myInt - 1)
    println(myInt) // 10

    var myInt2 : Int = 5
    println(myInt2)

    // 3. Variable de tipo decimal (Double, Float)

    var myDouble = 6.5
    println(myDouble)
    myDouble = 6.0
    println(myDouble) // 6.0. Se ha cambiado el valor

    var myDouble2: Double = 7.5
    var myFloat: Float = 8.0f 

    // 4. Variable de tipo buleano (True - False)

    var myBool = false
    myBool = true
    println(myBool)

    // - Opcionales

    var myOptional: String? = null // Declaramos la variable como un posible 'String' pero al principio se le asigna 'null'.
    println(myOptional) // null
    myOptional = "Mi cadena de texto"
    println(myOptional) // Imprime el valor añadido

    // - Esto es propio de kotlin
    myOptional?.let {

    }


    // - Constantes. Se usa la palabra reservada 'val' para constantes.

    val myConst = "Una constante de tipo cadena"
    // myConst = "Tratando de cambiar valor del constante" // Error.


    // - Interpolacion
    println("Este es el valor de la variable myString: " + myString)

    println("Este es el valor de la variable myInt: " $myInt)


    // - Documentacion:
    // kotlinlang.org/docs/home.html
    // developer.android.com/kotlin
    // jetbrains.com/pages/academy/kotlin

}

File: 02_Lenguajes_Programacion/11_Swift/1_sintaxis_swift/1_introduccion.swift
// Lenguaje Swift

// - Creado en Junio de 2014 por Apple. Soporta multiples paradigmas, orientada a protocolos, orientado a objetos, funcional, imperativo, estructura de bloque, declarativo y concurrente.
// - Sirve para la creacion de aplicaciones en MacOs, iOS, iPad y los demás dispositivos de la marca Apple.


//import UIKit // Para interfaces gráficas
import Foundation // núcleo del lenguaje Swift

print("Hola mundo en swift")


// Variables. Se puede declarar el tipo de dato pero el lenguaje puede inferirlo.

// Tipo Cadena (String)

var myString = "Esto es una cadena de texto"
myString = "Cambiando valor de la cadena"
// myString = 6 // Error

print(myString)

var myString2:String = "Otra cadena de texto"
print(myString2)

// Tipo Entero (Int)

var myInt = 6
myInt + 4
print(myInt) // 10
print(myInt - 1) // 11

var myInt2: Int = 5
print(myInt2)

// Tipo Decimal (Double, Float se debe definir, etc)

var myDouble = 6.5
print(myInt2)

var myFloat: Float = 8.3
print(myFloat)

// Tipo Buleano (Boolean)

var myBool = false
myBool = true // Se cambia el tipo de buleano
print(myBool)

// Constantes

let myConst = "Valor constante"
// myConst = "Modificando constante" // Error
print(myConst)
